platform/sync-installer/sync-installer-impl/src/main/java/org/codice/ddf/sync/installer/impl/SynchronizedInstallerImpl.java

```diff
@@ -0,0 +1,598 @@
+      throws InterruptedException {
+    ServiceTracker st = getServiceTracker(bundleContext, getServiceIdFilter(servicePid));
+    st.open(true);
+
+    Callable<Boolean> isServiceAvailable =
+        () -> {
+          LOGGER.trace("Waiting for service with pid [{}] to be registered", servicePid);
+          return st.getService() != null;
+        };
+
+    try {
+      wait(
+          isServiceAvailable,
+          maxWaitTime,
+          DEFAULT_POLLING_INTERVAL,
+          "Managed service failed to appear after [" + maxWaitTime + "] ms.");
+    } finally {
+      st.close();
+    }
+  }
+
+  @Override
+  public void installFeatures(String feature, String... additionalFeatures)
+      throws InterruptedException {
+    installFeatures(EnumSet.noneOf(FeaturesService.Option.class), feature, additionalFeatures);
+  }
+
+  @Override
+  public void installFeatures(
+      EnumSet<FeaturesService.Option> options, String feature, String... additionalFeatures)
+      throws InterruptedException {
+    installFeatures(DEFAULT_MAX_FEATURE_WAIT, options, feature, additionalFeatures);
+  }
+
+  @Override
+  public void installFeatures(
+      long maxWaitTime,
+      EnumSet<FeaturesService.Option> options,
+      String feature,
+      String... additionalFeatures)
+      throws InterruptedException {
+    Set<String> featuresToInstall =
+        featuresFromNames(feature, additionalFeatures)
+            .filter(f -> !featuresService.isInstalled(f))
+            .map(Feature::getName)
+            .collect(Collectors.toSet());
+    String featureNames = String.join(", ", featuresToInstall);
+
+    LOGGER.info("Installing the following features: [{}]", featureNames);
+    long startTime = System.currentTimeMillis();
+
+    try {
+      featuresService.installFeatures(featuresToInstall, options);
+    } catch (Exception e) {
+      throw new SynchronizedInstallerException(
+          "Failed to install features [" + String.join(",", featuresToInstall) + "]", e);
+    }
+
+    waitForBundles(getRemainingTime(startTime, maxWaitTime));
+    LOGGER.info("Finished installing features in {} ms", (System.currentTimeMillis() - startTime));
+  }
+
+  @Override
+  public void uninstallFeatures(String feature, String... additionalFeatures)
+      throws InterruptedException {
+    uninstallFeatures(EnumSet.noneOf(FeaturesService.Option.class), feature, additionalFeatures);
+  }
+
+  @Override
+  public void uninstallFeatures(
+      EnumSet<FeaturesService.Option> options, String feature, String... additionalFeatures)
+      throws InterruptedException {
+    uninstallFeatures(DEFAULT_MAX_FEATURE_WAIT, options, feature, additionalFeatures);
+  }
+
+  @Override
+  public void uninstallFeatures(
+      long maxWaitTime,
+      EnumSet<FeaturesService.Option> options,
+      String feature,
+      String... additionalFeatures)
+      throws InterruptedException {
+    Set<String> featuresToUninstall =
```

Comment:
❓ What happens if the resulting set is empty?

Reply:
Nothing, it does take the features service a bit longer to return though so I added a check.